test/sw_mgmt_spec.rb

```diff
@@ -46,6 +46,24 @@
     }
   end
 
+  describe ".init" do
+    it "initializes package management and returns true" do
+      expect(Yast::PackageLock).to receive(:Connect).with(false).and_return("connected" => true)
+
+      expect(Yast::PackageCallbacks).to receive(:InitPackageCallbacks)
+      expect(Yast::Pkg).to receive(:TargetInitialize)
+      expect(Yast::Pkg).to receive(:TargetLoad)
+      expect(Yast::Pkg).to receive(:SourceRestore).and_return(true)
+
+      expect(Registration::SwMgmt.init).to eq(true)
+    end
+
+    it "returns false when cannot obtain libzypp lock" do
+      expect(Yast::PackageLock).to receive(:Connect).with(false).and_return("connected" => false)
```

Comment:
Oh, I didn't know that it's possible to use inside a `before` block a value which is defined later inside `context` block, interesting... :astonished: